File: app/src/main/java/com/sumian/sleepdoctor/notification/NotificationListContract.java
package com.sumian.sleepdoctor.notification;

import com.sumian.sleepdoctor.base.SdBasePresenter;
import com.sumian.sleepdoctor.base.SdBaseView;
import com.sumian.sleepdoctor.notification.bean.Notification;

import java.util.List;

/**
 * <pre>
 *     @author : Zhan Xuzhao
 *     e-mail : xuzhao.z@sumian.com
 *     time   : 2018/6/6 16:14
 *     desc   :
 *     version: 1.0
 * </pre>
 */
class NotificationListContract {

    interface Presenter extends SdBasePresenter {
        void loadMore();

        void readNotification(String notificationId);
    }

    interface View extends SdBaseView<Presenter> {
        void onLoadMore(List<Notification> notificationList, boolean hasMore);

        void onReadSuccess();
    }

}

File: doctor/src/main/java/com/sumian/sddoctor/notification/NotificationListContract.kt
package com.sumian.sddoctor.notification

import com.sumian.sddoctor.base.BasePresenter
import com.sumian.sddoctor.base.BaseView
import com.sumian.sddoctor.notification.bean.Notification

/**
 * <pre>
 * @author : Zhan Xuzhao
 * e-mail : xuzhao.z@sumian.com
 * time   : 2018/6/6 16:14
 * desc   :
 * version: 1.0
</pre> *
 */
internal class NotificationListContract {
    internal interface View : BaseView {
        fun onLoadMore(notificationList: List<Notification>, hasMore: Boolean)

        fun onReadSuccess(notificationId: String, position: Int)
    }

    internal interface Presenter : BasePresenter {
        fun loadMore()

        fun readNotification(notificationId: String, notificationDataId: Int, position: Int)
    }
}
